lib/modules/manager/pub/artifacts.ts

```diff
@@ -0,0 +1,98 @@
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import {
+  getSiblingFileName,
+  readLocalFile,
+  writeLocalFile,
+} from '../../../util/fs';
+import { regEx } from '../../../util/regex';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+export async function updateArtifacts({
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  if (
+    !(
+      config.updateType === 'lockFileMaintenance' ||
+      config.updateType === 'lockfileUpdate'
+    ) &&
+    updatedDeps.length < 1
+  ) {
+    return null;
+  }
+
+  const lockFileName = getSiblingFileName(packageFileName, 'pubspec.lock');
+  const oldLockFileContent = await readLocalFile(lockFileName, 'utf8');
+  if (!oldLockFileContent) {
+    return null;
+  }
+
+  try {
+    await writeLocalFile(packageFileName, newPackageFileContent);
+
+    const isFlutter = newPackageFileContent.includes('sdk: flutter');
+    const toolName = isFlutter ? 'flutter' : 'dart';
+
+    let constraint: string | null;
+    if (isFlutter) {
+      const match = regEx(/^\tflutter: ['|"](?<flutterVersion>.*)['|"]$/m).exec(
```

Comment:
```suggestion
      const match = regEx(/^\tflutter: ['"](?<flutterVersion>.*)['"]$/m).exec(
```
i don't think the version can be quoted with a pipe `|`

Reply:
ah yes my stupid mistake, i thought i was in a group